extensions/mssql/src/objectExplorerNodeProvider/webhdfs.ts

```diff
@@ -67,7 +67,7 @@ export class WebHDFS {
 			params || {}
 		);
 		endpoint.search = querystring.stringify(searchOpts);
-		return url.format(endpoint);
+		return encodeURI(url.format(endpoint));
```

Comment:
Cool! Are there any cases where this could mess things up by double-encoding things?

@YurongHe, I would like to recommend adding tests for this, but see that the unit tests in the SQL2019 extension (e.g. hdfs.test.ts) weren't moved over. Do we have a task to track moving this and/or doing integration tests? If we can create tracking issues for moving the tests over in general, plus getting a specific integration test for this, that'd be great. if it's at all possible to add in the integration test to this review I'd be a very happy engineer.

Reply:
It shouldn't because the encodeURI will handle already encoded ones gracefully. But I'll test for confirming. :)
I'll work with Yurong on adding integration tests for this.

Reply:
@kevcunnane  encodeURI would mess up if we call it twice 👎. For now I checked the code under objectExplorerNodeProvider and we aren't using it anywhere and the place I added would handle encoding for all the webhdfs requests. I checked online but there isn't an elegant way to check if the url is already encoded.